pallets/moonbeam-foreign-assets/src/lib.rs

```diff
@@ -754,4 +778,65 @@ pub mod pallet {
 			AssetsById::<T>::get(asset_id)
 		}
 	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	impl<T: Config> SimpleCreate for Pallet<T> {
+		fn create_asset(id: u128, xcm_location: Location) -> DispatchResult {
+			use frame_support::traits::OriginTrait;
+			use sp_std::vec;
+			Pallet::<T>::create_foreign_asset(
+				<T as frame_system::Config>::RuntimeOrigin::root(),
+				id,
+				xcm_location,
+				12,
+				vec![b'M', b'T'].try_into().unwrap(),
+				vec![b'M', b'y', b'T', b'o', b'k'].try_into().unwrap(),
+			)
+		}
+	}
+
+	impl<T: Config> SimpleMutate<T> for Pallet<T> {
+		fn transfer_asset(
+			id: u128,
+			from: T::AccountId,
+			to: T::AccountId,
+			amount: U256,
+		) -> DispatchResult {
+			Pallet::<T>::transfer(id, from, to, amount)
+				.map_err(|_| Error::<T>::EvmCallTransferFail)?;
+			Ok(())
+		}
+
+		#[cfg(feature = "runtime-benchmarks")]
+		fn mint_asset(id: u128, account: T::AccountId, amount: U256) -> DispatchResult {
+			Pallet::<T>::mint_into(id, account, amount)?;
+			Ok(())
+		}
+	}
+
+	impl<T: Config> SimpleAssetExists for Pallet<T> {
+		fn asset_exists(id: u128) -> bool {
+			AssetsById::<T>::contains_key(&id)
+		}
+	}
+}
+
+pub trait SimpleCreate {
+	fn create_asset(id: u128, xcm_location: Location) -> DispatchResult;
+}
+
+pub trait SimpleMutate<R: frame_system::Config> {
+	fn transfer_asset(
+		id: u128,
+		from: R::AccountId,
+		to: R::AccountId,
+		amount: U256,
+	) -> DispatchResult;
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn mint_asset(id: u128, account: R::AccountId, amount: U256) -> DispatchResult;
+}
+
+pub trait SimpleAssetExists {
+	fn asset_exists(id: u128) -> bool;
 }
```

Comment:
Why do we need this? It only adds complexity

Reply:
Without an trait interface, `xcm-weight-trader` depends on the whole `moonbeam-foreign-assets` pallet, which makes, among other things, its own tests too convoluted without reason. Besides @stiiifff comments, this was introduced to make it less complex to work with.

I'm not quite sure about having the traits in this pallet, but besides that, it helped reduce duplicated and unused code.

Reply:
Why would `xcm-weight-trader` depend on `moonbeam-foreign-assets`? `xcm-weight-trader` is only used to keep the price of locations relative to the native asset

Reply:
For `xcm-weight-trader` to be able to convert between an AssetId and a relative price to the Native Asset it needs something that provides a mapping between `id` and `Location`, `moonbeam-foreign-assets` provides that mapping.

Regarding benchmarks, the `treasury` benchmarks requires that whatever implements the conversion trait also provides some way to ensure a conversion will be successful during benchmarks, this is why `xcm-weight-trader` depends again on `moonbeam-foreign-assets` to create an asset in order to add it to its support assets list.

Reply:
The traits are also used by the paymaster for dealing with transfers and minting assets to support the `treasury` benchmarks.

Reply:
`ConversionFromAssetBalance` does not need to be implemented in `xcm-weight-trader`. We can have a new struct implementing `ConversionFromAssetBalance` in the runtime (eg. inside `runtime/common`). This acts like a broker, connecting all components without making them depending on each other.

Reply:
Ended up extracting the Conversion into its own struct :+1: